plugins/modules/swim_workflow_manager.py

```diff
@@ -620,6 +703,46 @@ def get_image_id(self, name):
 
         return image_id
 
+    def get_cco_image_id(self, cco_image_name):
+        """
+    Retrieve the unique image ID from Cisco.com based on the provided image name.
+
+    Parameters:
+        self (object): An instance of a class used for interacting with Cisco DNA Center.
+        cco_image_name (str): The name of the software image to search for on Cisco.com.
+
+    Returns:
+        str: The image ID corresponding to the given image name.
+
+    Raises:
+        AnsibleFailJson: If the image ID cannot be found in the response.
+
+    Description:
+        This function sends a request to Cisco DNA Center to retrieve a list of software images
+        using the `returns_list_of_software_images` API. It then iterates through the response
+        to find a match for the provided `cco_image_name`. If a match is found, the corresponding
+        image ID is returned. If no matching image is found, or if the image ID is not present
+        in the response, the function logs an error message and raises an exception.
+    """
+
+        response = self.dnac._exec(
+            family="software_image_management_swim",
+            function='returns_list_of_software_images',
+            op_modifies=True,
+        )
+        self.log("Received API response from {0}: {1}".format("returns_list_of_software_images", (response)), "DEBUG")
+        response = response.get("response")
+
```

Comment:
Add the check if you receive the epmty response.
```
if not response:
    ....

Reply:
updated the code

Reply:
when we give like if not response the code is not getting worked as intended because response will never be empty we try this way but the code is not working